src/Exercism.TestRunner.CSharp/TestsRewriter.cs

```diff
@@ -204,7 +204,7 @@ public override SyntaxNode VisitClassDeclaration(ClassDeclarationSyntax node)
                                 IdentifierName("IDisposable"))));
 
                 // TODO: refactor to method
-                if (node.DescendantNodes().OfType<MethodDeclarationSyntax>().All(method => method.Identifier.Text != "Dispose"))
+                if (node.DescendantNodes().Where(n => n.Parent?.Equals(node) ?? false).OfType<MethodDeclarationSyntax>().All(method => method.Identifier.Text != "Dispose"))
```

Comment:
Could you see if maybe the fix could be to use `ChildNodes()`? (see https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.syntaxnode.childnodes?view=roslyn-dotnet-3.11.0)

```suggestion
                if (node.ChildNodes().OfType<MethodDeclarationSyntax>().All(method => method.Identifier.Text != "Dispose"))
```

Reply:
Yes, it seems to be the right solution. I was looking for something like this, but in the wrong place as it turns out.